db/bigtable_common.go

```diff
@@ -0,0 +1,133 @@
+package db
+
+import (
+	"context"
+	"eth2-exporter/types"
+	"eth2-exporter/utils"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	gcp_bigtable "cloud.google.com/go/bigtable"
+)
+
+func (bigtable *Bigtable) WriteBulk(mutations *types.BulkMutations, table *gcp_bigtable.Table, batchSize int) error {
```

Comment:
**Question:** Having two `for` loops here is odd and feels error prone as you would have to make changes in two places. Why not have a single `for` loop?

Reply:
I don't understand the question

Reply:
This is because I wrote it wrong, sorry. Here is the fixed version:

---

**Question:** Having two parts calling `ApplyBulk` instead of doing everything within the `for` loop feels error prone as you would have to make changes in two places whenever changes are needed. Why not have everything related to `ApplyBulk` within the `for` loop and get rid of this part:
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/507bb0a4ac1516f0eb939147e4a349bae1f10bf4/db/bigtable_common.go#L50-L67

Reply:
As discussed in DMs: Won't be changed.